OpenEdXMobile/src/main/java/org/edx/mobile/view/MainDiscoveryFragment.java

```diff
@@ -130,13 +130,16 @@ private void initFragments() {
     }
 
     private void onFragmentSelected(@IdRes int resId, final boolean isUserSelected) {
+        Fragment selectedFragment = null;
         for (int i = 0; i < fragmentsArray.size(); i++) {
             if (resId == fragmentsArray.keyAt(i)) {
-                showFragment(fragmentsArray.valueAt(i));
+                selectedFragment = fragmentsArray.valueAt(i);
             } else {
                 hideFragment(fragmentsArray.valueAt(i));
             }
         }
+        if (selectedFragment != null)
```

Comment:
A block comment above this block of code would be great that explains why we are first hiding all fragments and showing one at the end.